src/exceptions.py

```diff
@@ -38,6 +38,22 @@ def __init__(self, msg: str):
         self.msg = msg
 
 
+class CharmDatabaseRelationNotFoundError(Exception):
```

Comment:
In line with the comment above, I think `CommandMigrateConfigError` and `ServerNameModifiedError` belong in synapse.py as they are part of the bussiness logic this file encapsultes.
`CharmConfigInvalidError` would probably belong in charm_state.py

Reply:
Nice, changed.